src/jit/emitarm64.cpp

```diff
@@ -7977,120 +8101,290 @@ BYTE*               emitter::emitOutputLJ(insGroup  *ig, BYTE *dst, instrDesc *i
     }
 #endif
 
-    loBits = (distVal & 3);
-    distVal >>= 2; // branch offset encodings are scaled by 4.
-
     /* For forward jumps, record the address of the distance value */
     id->idjTemp.idjAddr = (distVal > 0) ? dst : NULL;
 
-    assert(emitJumpCrossHotColdBoundary(srcOffs, dstOffs) == false);
+    if (emitJumpCrossHotColdBoundary(srcOffs, dstOffs))
+    {
+        assert(!id->idjShort);
+        NYI_ARM64("Relocation Support for long address");
+    }
+
     assert(insOptsNone(id->idInsOpt()));
 
     if (isJump)
     {
-        assert(!id->idjKeepLong);
-
-        // branch offsets must be a multiple of 4
-        noway_assert(loBits == 0);
-
-        if (fmt == IF_BI_0A)
-        {
-            // INS_b or INS_bl_local
-            noway_assert(isValidSimm26(distVal));
-            distVal &= 0x3FFFFFFLL;
-            code |= distVal;
-        }
-        else if (fmt == IF_BI_0B)
+        if (id->idjShort)
         {
-            // INS_beq, INS_bne, etc...
-            noway_assert(isValidSimm19(distVal));
-            distVal &= 0x7FFFFLL;
-            code |= distVal << 5;
+            // Short conditional/unconditional jump
+            assert(!id->idjKeepLong);
+            assert(emitJumpCrossHotColdBoundary(srcOffs, dstOffs) == false);
+            assert((fmt == IF_BI_0A) || (fmt == IF_BI_0B));
         }
-        else if (fmt == IF_BI_1A)  // BI_1A   X.......iiiiiiii iiiiiiiiiiittttt      Rt simm19:00
+        else
         {
-            // INS_cbz or INS_cbnz
-            code |= insEncodeDatasize(id->idOpSize());    // X 
-            code |= insEncodeReg_Rt(id->idReg1());        // ttttt
+            // Long conditional jump
+            assert(fmt == IF_LARGEJMP);
+            // This is a pseudo-instruction format representing a large conditional branch, to allow
+            // us to get a greater branch target range than we can get by using a straightforward conditional
+            // branch. It is encoded as a short conditional branch that branches around a long unconditional
+            // branch.
+            //
+            // Conceptually, we have:
+            //
+            //      b<cond> L_target
+            //
+            // The code we emit is:
+            //
+            //      b<!cond> L_not  // 4 bytes. Note that we reverse the condition.
+            //      b L_target      // 4 bytes
+            //   L_not:
+            //
+            // Note that we don't actually insert any blocks: we simply encode "b <!cond> L_not" as a branch with
+            // the correct offset. Note also that this works for both integer and floating-point conditions, because
+            // the condition inversion takes ordered/unordered into account, preserving NaN behavior. For example,
+            // "GT" (greater than) is inverted to "LE" (less than, equal, or unordered).
+            //
+            // History: previously, we generated:
```

Comment:
This History refers to what we did for Arm32,  (the it<cond> instruction)
I'm not sure why we want to mention it here in the Arm64 implementation